src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
This should indeed work for move _across_ the tree.
I am not sure how this would behave for sub tree moving. But that depends on solving #11 first anyway.